File: Swift/프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.swift
import Foundation

func solution(_ s:String) -> Int {
    
    let numberSet: Set<String> = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"]
    let numDic: [String : String] = ["zero" : "0", "one" : "1", "two" : "2", "three" : "3", "four" : "4", "five" : "5", "six" : "6", "seven" :  "7", "eight" : "8", "nine" : "9"]
    
    var resultString = ""
    var tempString = ""
    for i in 0..<s.count {
        let index = s.index(s.startIndex, offsetBy: i)
        let char = String(s[index])
        if numberSet.contains(char) {
            resultString += char
        } else {
            tempString += char
            if let number = numDic[tempString] {
                resultString += number
                tempString = ""
            }
        }
    }
    
    return Int(resultString)!
}

File: Java/프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.HashMap;

class Solution {
    public int solution(String s) {
        HashMap<String, String> pair = new HashMap<>();
        
        pair.put("zero", "0");
        pair.put("one", "1");
        pair.put("two", "2");
        pair.put("three", "3");
        pair.put("four", "4");
        pair.put("five", "5");
        pair.put("six", "6");
        pair.put("seven", "7");
        pair.put("eight", "8");
        pair.put("nine", "9");
        
        int sCount = s.length();
        
        String tempString = "";
        String resultString = "";
        for (int i = 0; i<sCount; i++) {
            tempString += s.charAt(i);
            if (pair.containsValue(tempString)) {
                resultString += tempString;
                tempString = "";
            } else if (pair.containsKey(tempString)) {
                resultString += pair.get(tempString);
                tempString = "";
            }
        }
        
        
        return Integer.parseInt(resultString);
    }
}